src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -229,7 +229,7 @@ public static CodeWriter WriteParametersValidation(this CodeWriter writer, IEnum
 
         private static CodeWriter WriteParameterValidation(this CodeWriter writer, Parameter parameter)
         {
-            if (parameter.Initializer != null)
+            if (parameter.Validation == ValidationType.None && parameter.Initializer != null)
```

Comment:
Why do you need this change?

Reply:
E.g., 
```
public AccessibilityClient(AzureKeyCredential credential, Uri endpoint, AccessibilityClientOptions options)
{
    Argument.AssertNotNull(credential, nameof(credential));
    Argument.AssertNotNull(endpoint, nameof(endpoint));
    options ??= new AccessibilityClientOptions();
}
```
`options` and `endpoint` are both required, but we want to validate `endpoint` notnull while initialize `options` by default value. So when I constructor the parameters, I make the `validation` of `options` as `none`.